packages/core/src/logger/apiLogger.ts

```diff
@@ -0,0 +1,260 @@
+      }
+    );
+
+    this.applyLogResponseOptions(logLevel, response);
+  }
+
+  private applyLogRequestOptions(level: Level, request: HttpRequest) {
+    this.applyLogRequestHeaders(
+      level,
+      request,
+      this._loggingOptions.logRequest
+    );
+
+    this.applyLogRequestBody(level, request, this._loggingOptions.logRequest);
+  }
+
+  private applyLogRequestHeaders(
+    level: Level,
+    request: HttpRequest,
+    logRequest?: HttpRequestLoggingOptions
+  ) {
+    if (logRequest) {
+      const { logHeaders, headerToInclude, headerToExclude } = logRequest;
+
+      if (logHeaders) {
+        const headersToLog = this.extractHeadersToLog(
+          request.headers,
+          headerToInclude,
+          headerToExclude
+        );
+
+        this._loggingOptions.logger?.log(level, 'Request Headers: ${headers}', {
+          headers: headersToLog,
+        });
+      }
+    }
+  }
+
+  private applyLogRequestBody(
+    level: Level,
+    request: HttpRequest,
+    logRequest?: HttpRequestLoggingOptions
+  ) {
+    if (logRequest?.logBody) {
+      this._loggingOptions.logger?.log(level, 'Request Body: ${body}', {
+        body: request.body,
+      });
+    }
+  }
+
+  private applyLogResponseOptions(level: Level, response: HttpResponse) {
+    this.applyLogResponseHeaders(
+      level,
+      response,
+      this._loggingOptions.logResponse
+    );
+
+    this.applyLogResponseBody(
+      level,
+      response,
+      this._loggingOptions.logResponse
+    );
+  }
+
+  private applyLogResponseHeaders(
+    level: Level,
+    response: HttpResponse,
+    logResponse?: HttpMessageLoggingOptions
+  ) {
+    if (logResponse) {
+      const { logHeaders, headerToInclude, headerToExclude } = logResponse;
+
+      if (logHeaders) {
+        const headersToLog = this.extractHeadersToLog(
+          response.headers,
+          headerToInclude,
+          headerToExclude
+        );
+
+        this._logger.log(level, 'Response Headers: ${headers}', {
+          headers: headersToLog,
+        });
+      }
+    }
+  }
+
+  private applyLogResponseBody(
+    level: Level,
+    response: HttpResponse,
+    logResponse?: HttpMessageLoggingOptions
+  ) {
+    if (logResponse?.logBody) {
+      this._logger.log(level, 'Response Body: ${body}', {
+        body: response.body,
+      });
+    }
+  }
+
+  private getContentType(headers?: Record<string, string>): string {
+    return headers ? getHeader(headers, CONTENT_TYPE_HEADER) ?? '' : '';
+  }
+
+  private getContentLength(headers?: Record<string, string>): string {
+    return headers ? getHeader(headers, CONTENT_LENGTH_HEADER) ?? '' : '';
+  }
+
+  private removeQueryParams(url: string): string {
+    const queryStringIndex: number = url.indexOf('?');
+    return queryStringIndex !== -1 ? url.substring(0, queryStringIndex) : url;
+  }
+
+  private extractHeadersToLog(
+    headers?: Record<string, string>,
+    headersToInclude?: string[],
+    headersToExclude?: string[]
```

Comment:
There won't be any need to have this parameter as optional once you fix the issue I mentioned in the logging options interfaces.